pandas/core/indexing.py

```diff
@@ -693,8 +694,23 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None):
             # GH#38148
             keys = self.obj.columns.union(key, sort=False)
 
+            # Try to get the right dtype when we do this reindex.
```

Comment:
doesn't infer_dtype_from do this?

Reply:
that infers a dtype, but we need to pass a scalar fill_value to reindex_axis

Reply:
removed this edit as no longer necessary.  i think at this point all the controversial bits are gone and we're down to just the bugfix.